### PR TITLE
Scene assets load update

### PR DESCRIPTION
**Pull request by makiam**
Scene assets load update. Use common method to load both materials and textures

### CODE REVIEW

**Comment by peastman** [comment]
Could you explain the design and reasons for this change?  What is `setNameFluent()` for?

**Comment by makiam** [comment]
This is a just a syntax sugar to write object initialization code this way..
>>`Supplier<Named>> stub`
...
>> `stub.get().setNameFluent('<unreadable>')`

**Comment by peastman** [comment]
What is "fluent" supposed to mean?

**Comment by makiam** [comment]
https://en.wikipedia.org/wiki/Fluent_interface

File: ArtOfIllusion/src/artofillusion/Scene.java [hunk 1]
```diff
@@ -1208,8 +1209,59 @@ public Scene(DataInputStream in, boolean fullScene) throws IOException, InvalidO
...
+          result.add((Named)stub.get().setNameFluent("<unreadable>"));
+          scene.errors.add(Translate.text("errorInstantiatingClass", className));
+        }
+      }
+      return result;
+  }
+
   /** Initialize the scene based on information read from an input stream. */
 
+  @SuppressWarnings("UseOfObsoleteCollectionType")
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/Scene.java:1264]: Whatever else we decide about this change, please reserve `@SuppressWarnings` for instances where we *cannot* change to a newer collection class/structure/safe cast/etc.

And even then, It should probably go at the line that uses the 'iffy' code, with a comment that explains *why* we have to stick to the old way...

**Comment by Sailsman63** [comment]
Alright, finally got some time to consider this.

- Fluent: I kind of suspect that most fluent interfaces don't actually have "Fluent" in their names. An example of a fluent interface in java would be `java.util.stream.Stream`, in particular all of the methods that return a `Stream` (And arguably, those that return one of the specialized Stream variants) 
- I can see that you are trying to unify the loading of assets, but this `Named` Interface just seems clumsy. It's not in itself a self-explaining abstraction, especially when it's bolted on this way.
- If we were to try to unify the loading of assets, we're essentially saying that `Textures`, `Materials` or any other asset type are basically always mirrors of each other. This may not always be the case.

**Comment by makiam** [comment]
For now I use setNameFluent because changing setName method return type gives me conflict with ProcedureOwner interface which also declares name getter and setter methods. In general ProcedureOwner looks responsible for many things which is not good IMO.

**Comment by makiam** [comment]
For future...  Apply Named interface for all objects with name getter/setters.

**Comment by Sailsman63** [comment]
Let me phrase this another way:

What purpose does the abstraction "named" serve? Is there any case where we would want to operate on a collection of "named" things, without considering what else they might be?


If you've got concerns about `ProcedureOwner` in general, please open a separate issue to explain them.

**Comment by makiam** [comment]
Every object marked with this interface may return it's name and can change it (Which is used in this particular case)

**Comment by Sailsman63** [comment]
I think we're having a serious communication issue here...

What makes this significant enough to justify a separate interface? Again, is there any use distinguishing between items that are "named" and those that are not?

**Comment by makiam** [comment]
Old code for loading textures/materials has 2 points where object name are set. (One for fallback texture and one for material). Once I write common code to load both asset types I need common denominator to set name. The Named interface serves as this denominator.

**Comment by peastman** [comment]
I don't think I've ever encountered a method with the name "fluent" in it before.  Most people will have no idea what it refers to.  I certainly didn't.  And if you want an API where calls can be chained together like that, you need to make all setter methods do it.  Having just one that can be chained when none of the others can doesn't help anyone.  It just makes the API confusing.

> Old code for loading textures/materials has 2 points where object name are set.

I could see an argument for creating a common interface implemented by both textures and materials.  Names are only a small part of what they have in common.  They also have IDs, can use ImageMaps, can be saved to files, etc.  If we want to consider that, let's enumerate everything that interface might include.  Then we need to figure out whether there's actually enough benefit to justify it.  If you almost always know whether you're working with a Texture or a Material, then it doesn't make much difference.  But if there's a lot of code that would benefit from being able to accept either Textures or Materials, then it might make sense.

**Comment by Sailsman63** [comment]
These all got duplicated in #212, continue any discussion over there.

**Pull request closed without merging**